### PR TITLE
[F2F-1118] - Improve Test Coverage

### PR DESCRIPTION
**Pull request by mcgarj**
### What changed

Unit tests were added to controllers and other files

Screenshot 2023-08-21 at 10 46 20

### Why did it change

Test coverage was below 60% expected for the FE

### Issue tracking

- [F2F-1118](https://govukverify.atlassian.net/browse/F2F-1118)

[F2F-1118]: https://govukverify.atlassian.net/browse/F2F-1118?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ViveakV — DISMISSED** [review]

**Comment by ViveakV** [comment]
Failing tests @mcgarj

File: src/app/f2f/controllers/checkDetails.test.js [hunk 1]
```diff
@@ -76,8 +76,40 @@ describe("CheckDetails controller", () => {
...
+
+        expect(next).to.have.been.calledWith(error, superLocals);
+        expect(postOfficeName).to.equal("BRANCH NAME");
+        expect(postOfficePostcode).to.equal("G41 1ED");
+        expect(postCodeLat).to.equal("100000");
+        expect(postCodeLong).to.equal("-100000");
+        expect(countryCode).to.equal("GBR");
+        expect(country).to.equal("United Kingdom");
+        
+      });
```
Comment (Caitlin-cooling) [inline src/app/f2f/controllers/checkDetails.test.js:97]: Do we want a test for the second option?

Also, can I ask why the test for the first option is so different from the ones for the third and last?

File: src/app/f2f/controllers/landingPage.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  it("should be an instance of BaseController", () => {
+    expect(landingPageController).to.be.an.instanceOf(BaseController);
+  });
+
+  describe("#saveValues", () => {
+    it("should save the values to the sessionModel", async () => {
+      req.session.tokenId = "123abc"
+
+      await landingPageController.saveValues(req, res, next);
+      expect(next).to.have.been.calledOnce;
```
Comment (Caitlin-cooling) [inline src/app/f2f/controllers/landingPage.test.js:30]: There's quite a lot of other logic in the `saveValues` function (eg setting `isThinFileUser` to false, fetching the session config, etc.) is there a reason that we haven't tested it?

File: src/app/f2f/countryCodes/euDrivingLicence.test.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const { EU_DL_COUNTRIES } = require("./euDrivingLicence");
+
+describe("#EU_DL_COUNTRIES", () => {
+  it("Should contain nested objects with 2 values - text and code as strings", () => {
+    expect(EU_DL_COUNTRIES.DENMARK).to.deep.equal({ text: "Denmark", code: "DNK"} )
```
Comment (Caitlin-cooling) [inline src/app/f2f/countryCodes/euDrivingLicence.test.js:5]: Snapshot testing https://jestjs.io/docs/snapshot-testing might be a better alternative to just testing one country - what do you think?

File: src/app/f2f/controllers/landingPage.test.js (hunk 2 above)
Comment (mcgarj) [reply src/app/f2f/controllers/landingPage.test.js:30]: It's mainly time constraints honestly. I picked the easiest things to test to get the coverage over 60% (it should be around 70% now)

File: src/app/f2f/countryCodes/euDrivingLicence.test.js (hunk 3 above)
Comment (mcgarj) [reply src/app/f2f/countryCodes/euDrivingLicence.test.js:5]: Could be a good idea!

**Review by Caitlin-cooling — APPROVED** [review]

**Pull request merged**